### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by RemLampa**

### CODE REVIEW

File: resources/boilerplates/flattenData.js [hunk 1]
```diff
@@ -15,10 +16,11 @@ export function flattenChildData(element) {
 export function flattenData(element) {
   const flatData = {
     ...element.instance,
+    children: []
   };
 
-  if (element.children) {
-    flatData.children = element.children.map(
+  if (element.children && element.children.length > 0) {
+    flatData.children = element.children.filter(child => !!child).map(
```
Comment (YizYah) [inline resources/boilerplates/flattenData.js]: Is it really necessary still for checking whether the length is greater than 0?  And to filter out the nulls?  We should be doing that...

File: resources/boilerplates/flattenData.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 export function flattenChildData(element) {
   const flatData ={
     typeId: element.typeId,
+    instances: []
   };
 
-  if (element.instances) {
-    flatData.instances = element.instances.map(
+  if (element.instances && element.instances.length > 0) {
+    flatData.instances = element.instances.filter(child => !!child).map(
```
Comment (YizYah) [inline resources/boilerplates/flattenData.js]: Is it really necessary to check whether children is empty?  Of course the map would return an empty array in such a case.  Also, are the nulls not being filtered out now by the back end?  They really should be I think. If not, that should be a fix.

File: resources/boilerplates/flattenData.js (hunk 1 above)
Comment (YizYah) [inline resources/boilerplates/flattenData.js]: Ditto

File: resources/boilerplates/genericPluralNonRoot.js [hunk 3]
```diff
@@ -6,19 +6,21 @@ import __SingularName__ from '../__SingularName__';
 
 const __PluralName__StyleWrapper = styled.div``;
 
-function __PluralName__({ __SingularNameLowercase__s, currentTodoId, onUpdate, refetchQueries }) {
+function __PluralName__({ __SingularNameLowercase__s, __SingularParentName__Id, currentTodoId, onUpdate, refetchQueries }) {
   return (
     <__PluralName__StyleWrapper>
       <__SingularName__CreationForm
-        currentTodoId={currentTodoId}
+        __SingularParentName__Id={__SingularParentName__Id}
```
Comment (YizYah) [inline resources/boilerplates/genericPluralNonRoot.js]: Ouch!

File: resources/boilerplates/flattenData.js (hunk 2 above)
Comment (RemLampa) [reply resources/boilerplates/flattenData.js]: Yeah, allow me to review this.

**Pull request merged**